src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,23 @@
+package seedu.addressbook.data.person;
+
+/**
+ * Represents an contact details.
+ */
+public abstract class Contact {
+    
+	protected boolean isPrivate;
+    protected final String value;
+
+    protected Contact(String value, boolean isPrivate) {
+    	this.value = value; 
+    	this.isPrivate = isPrivate; 
+    }
+	
+	public abstract String toString();
+	
+    public abstract int hashCode(); 
```

Comment:
If all child class has the same method, you can put the function here instead of making it `abstract`.